File: modules/common-net/src/main/java/com/shterneregen/securelan/common/net/transport/SocketClose.java
package com.shterneregen.securelan.common.net.transport;

public final class SocketClose {
    private SocketClose() {
    }

    public static void closeQuietly(AutoCloseable closeable) {
        if (closeable == null) {
            return;
        }
        try {
            closeable.close();
        } catch (Exception ignored) {
        }
    }
}

File: modules/common-net/src/main/kotlin/com/shterneregen/securelan/common/net/transport/SocketClose.kt
package com.shterneregen.securelan.common.net.transport

class SocketClose private constructor() {
    companion object {
        @JvmStatic
        fun closeQuietly(closeable: AutoCloseable?) {
            if (closeable == null) {
                return
            }
            try {
                closeable.close()
            } catch (_: Exception) {
            }
        }
    }
}
